language-support/ts/codegen/tests/ts/build-and-lint-test/src/__tests__/test.ts

```diff
@@ -611,3 +613,38 @@ test('package API', async () => {
   const downSuc = await ledger.getPackage(buildAndLint.packageId);
   expect(downSuc.byteLength > 0).toBe(true);
 });
+
+test('reconnect on timeout, when multiplexing is enabled', async () => {
+  const charlieLedger = new Ledger({token: CHARLIE_TOKEN, httpBaseUrl: httpBaseUrl(), multiplexQueryStreams: true});
+  const charlieRawStream = charlieLedger.streamQuery(buildAndLint.Main.Person, {party: CHARLIE_PARTY});
+  const charlieStream = promisifyStream(charlieRawStream);
+  const charlieStreamLive = pEvent(charlieRawStream, 'live');
+  expect(await charlieStreamLive).toEqual([]);
+
+  const charlieRecord1: buildAndLint.Main.Person = {
+    name: 'Charlie Chaplin',
+    party: CHARLIE_PARTY,
+    age: '10',
+    friends: [],
+  };
+
+  const charlieContract1 = await charlieLedger.create(buildAndLint.Main.Person, charlieRecord1);
+  expect(await charlieStream.next()).toEqual([[charlieContract1], [{created: charlieContract1, matchedQueries: [0]}]]);
+
+  // wait 70s to trigger a disconnect on json-api which is configured to close conn after 1 minute.
+  await new Promise(resolve => setTimeout(resolve, 70000));
```

Comment:
Can we reduce that a bit? IIRC the timeout on WS connection is configurable, it would be great if this could be set to like 10s for these tests here.

Reply:
IIUC `maxDuration` is specified in minutes, so 1 minute is the min we can do without changing the cli opt definition(might have to deprecate `maxDuration` in favour of a new flag `maxDurationSeconds`) or adding a hacky backdoor maybe via env variables to override this.
```scala 
protected[this] override def create(implicit readCtx: DummyImplicit) =
    for {
      md <- optionalLongField("maxDuration")
      hbp <- optionalLongField("heartBeatPer")
    } yield Config.DefaultWsConfig
      .copy(
        maxDuration = md
          .map(t => FiniteDuration(t, TimeUnit.MINUTES))
          .getOrElse(Config.DefaultWsConfig.maxDuration),
        heartBeatPer = hbp
          .map(t => FiniteDuration(t, TimeUnit.SECONDS))
          .getOrElse(Config.DefaultWsConfig.heartBeatPer),
      )
```

Reply:
Ugh. All right. 😿